crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space/rpc_multi_logic.rs

```diff
@@ -69,6 +69,29 @@ struct Outer {
     basis: Arc<KitsuneBasis>,
     payload: Vec<u8>,
 }
+struct Kill {
+    closed: AtomicBool,
+    kill: Notify,
+}
+
+impl Kill {
+    fn new() -> Arc<Self> {
+        Arc::new(Self {
+            closed: AtomicBool::new(false),
+            kill: Notify::new(),
+        })
+    }
+    fn kill_all(&self) {
+        self.closed
+            .store(true, std::sync::atomic::Ordering::Relaxed);
+        self.kill.notify_waiters();
+    }
+    async fn wait(&self) {
+        if !self.closed.load(std::sync::atomic::Ordering::Relaxed) {
+            self.kill.notified().await;
+        }
+    }
```

Comment:
I have to admit I don't fully grok the atomic orderings, but I have a vague suspicion this might be a case where we need SeqCst to ensure there isn't a race on checking closed right before it gets closed.

To further illuminate my non-understanding, we probably could set one to Acquire and one to Release, but I don't know which one should be which : )

Reply:
```suggestion
    fn kill_all(&self) {
        self.closed
            .store(true, std::sync::atomic::Ordering::SeqCst);
        self.kill.notify_waiters();
    }
    async fn wait(&self) {
        if !self.closed.load(std::sync::atomic::Ordering::SeqCst) {
            self.kill.notified().await;
        }
    }
```

Reply:
I was under the impression that a Relaxed ordering will always mean you read a write because it's only ever going from false to true but you need stronger ordering if it may change back and forth but I could definitely be wrong.

Reply:
I changed this to Acquire and Release after diving into the docs a bit. SeqCst seems to only be needed if you care about the order that multiple writes occurred but we only care that the load is acquired after the store is released.